src/herder/test/UpgradesTests.cpp

```diff
@@ -1949,7 +1949,7 @@ TEST_CASE("simulate upgrades", "[herder][upgrades][acceptance]")
                                                     {upgrade, genesis(0, 15)}};
         auto checks = std::vector<LedgerUpgradeCheck>{
             {genesis(0, 10), {noUpgrade, noUpgrade, noUpgrade}},
-            {genesis(0, 21), {upgrade, upgrade, upgrade}}};
+            {genesis(0, 28), {upgrade, upgrade, upgrade}}};
```

Comment:
what are those changes?

Reply:
The ones I described in this commit log:

> This required the timeouts on three upgrade tests to be
increased (because, I believe, of the extra signature-
checking during the nomination phase while running protocol
0).

Reply:
this doesn't make sense to me:
* we're in virtual time, so signature verification time is free
* in general, as the new behavior is supposed to pick (potentially) smaller close times, the ledger close time should maybe move in the other direction
* for this particular test, all nodes vote exactly the same than before, so I would expect the close times to be the same

I suggest you analyze what is really going on here...

Reply:
> in general, as the new behavior is supposed to pick (potentially) smaller close times, the ledger close time should maybe move in the other direction

I'll certainly look into this, but it has nothing to do with the new closetime behavior -- it's the pre-commit to remove support for unsigned nominations (as they were in protocols 10 and before) that induces this test timeout, without the closetime change being present yet.

Reply:
> we're in virtual time, so signature verification time is free

Ah, I figured the simulation must be charging us some time for verifying signatures.  If it doesn't, then yes, I am mystified.

Reply:
I think I understand what's happening here now.  I think that when we made nominations signed back in protocol 11, we may have increased the number of nomination messages.

The reason is that `NominationProtocol::processEnvelope()` tests whether a vote can be accepted using voted and accepted `StatementPredicate`s (parameters to `Slot::federatedAccept()`) which test whether a nomination is already present in our `st.pledges.nominate().votes` or `st.pledges.nominate().accepted` vectors by doing a std::find, which tests for strict equality.

Prior to the signing of nominations, therefore, we would treat a nomination as if we had already voted for (resp. accepted) it if we had already voted for (resp. accepted) a nomination with the same `txSetHash`, `closeTime`, and `upgrades`.

But with the signing of nominations, we will only treat a nomination as if we had already voted for (resp. accepted) it if we had already voted for (resp. accepted) a nomination which not only had the same `txSetHash`, `closeTime`, and `upgrades` , but also originated with the same node (and therefore had the same signature).

So it ends up taking more messages to reach consensus.

I have no idea whether that difference is significant in production (for all I know, it might be extremely rare in real life that different nodes would generate the same nominations), but it certainly affects the `UpgradesTests`, which like to set up several nodes that nominate the same stuff.

To confirm that that's what causes the timeouts, I wrote the following code, and it made them go away:

https://github.com/rokopt/stellar-core/commit/9bbeb4e04859a925779b4b219816d681a508efb8

However, I doubt that that's actually correct; it's just a test of the hypothesis.  It makes some other tests fail -- in particular the SCP tests that count `mEnvs` start getting the wrong counts.  I'm not sure whether we actually want to change anything there or not, and if we do want to do a node-insensitive comparison, I'm not sure exactly what we should do when we find that we've voted for/accepted a nomination that is equal except that it comes from a different node. 

The upshot as far as CAP-34 is concerned, though:  the root cause of these test failures is the introduction of signed nominations in protocol 11, not any of the code that's been changed in this PR.